### ISSUE TITLE
codes of MAD-X paper 

### ISSUE THREAD

**Issue by ghost** [Adapter-Hub/adapter-transformers#117]
Hi
could you assist me how I can get the implementation of MAD-X paper and how I can reproduce the results? I could not find the information, thanks

**Comment by calpt** [comment]
Hi @juliahane,

All implementation components of MAD-X have been integrated in this library. To get an overview on how to perform the cross-lingual transfer proposed in the paper, you can have a look at [this Colab notebook](https://colab.research.google.com/github/Adapter-Hub/adapter-transformers/blob/master/notebooks/04_Cross_Lingual_Transfer.ipynb). We have also modified a few training scripts from HuggingFace transformers to enable adapter training: [scripts](https://github.com/Adapter-Hub/adapter-transformers/tree/master/examples) / [documentation](https://docs.adapterhub.ml/training.html).

**Comment by juliahane** [comment]
Hi
thank you for the help, could you also direct me where I can get the codes
you used to train the language adapters? I am a bit confused on the
procedure to train adapterts, and very much appreciate sharing the codes to
train language adapters from unsupervised data. thanks

**Comment by JoPfeiff** [comment]
The wiki corpus was extracted using the [WikiExtractor library](https://github.com/attardi/wikiextractor)

Language adapters were trained on a single gpu with gradient accumulation.
